relay-cabi/src/processing.rs

```diff
@@ -9,29 +9,149 @@ use std::os::raw::c_char;
 use std::slice;
 use std::sync::OnceLock;
 
+use chrono::{DateTime, Utc};
 use relay_common::glob::{glob_match_bytes, GlobOptions};
 use relay_dynamic_config::{normalize_json, validate_json, GlobalConfig, ProjectConfig};
 use relay_event_normalization::{
-    normalize_event, validate_event_timestamps, validate_transaction, EventValidationConfig,
-    GeoIpLookup, NormalizationConfig, RawUserAgentInfo, StoreConfig, StoreProcessor,
+    normalize_event, validate_event_timestamps, validate_transaction, BreakdownsConfig,
+    ClientHints, EventValidationConfig, GeoIpLookup, NormalizationConfig, RawUserAgentInfo,
     TransactionValidationConfig,
 };
 use relay_event_schema::processor::{process_value, split_chunks, ProcessingState};
-use relay_event_schema::protocol::{Event, VALID_PLATFORMS};
+use relay_event_schema::protocol::{Event, IpAddr, VALID_PLATFORMS};
 use relay_pii::{
     selector_suggestions_from_value, DataScrubbingConfig, InvalidSelectorError, PiiConfig,
     PiiConfigError, PiiProcessor, SelectorSpec,
 };
 use relay_protocol::{Annotated, Remark, RuleCondition};
 use relay_sampling::SamplingConfig;
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
 
 use crate::core::{RelayBuf, RelayStr};
 
+/// Configuration for the store step -- validation and normalization.
+#[derive(Serialize, Deserialize, Debug, Default)]
+#[serde(default)]
+pub struct StoreConfig {
```

Comment:
I think we want to call this `StoreNormalizer` or wrap the config in a type called `StoreNormalizer` so we have still a consistent naming with the C-ABI functions.